src/Can/Handlers/CanDisplayPopupHandler2.h

```diff
@@ -0,0 +1,283 @@
+				return;
+		}
+
+		if (item.MessageType
+				== CAN_POPUP_MSG_AUTOMATIC_DOOR_LOCKING_ACTIVATED) {
+			if (GetEngineRunning() && !automaticDoorLockShownOnEngineRunning) {
+				PushPopupMsg(&item, canSemaphore, popupMessageQueue);
+				automaticDoorLockShownOnEngineRunning = true;
+			}
+
+			if (GetIgnition() && !automaticDoorLockShownOnIgnition) {
+				PushPopupMsg(&item, canSemaphore, popupMessageQueue);
+				automaticDoorLockShownOnIgnition = true;
+				itemdoor = item;
+
+			}
+
+			if (automaticDoorLockShownOnEngineRunning)
+				return;
+			if (automaticDoorLockShownOnIgnition)
+				return;
+		}
+
+		if ((millis() - previousCanPopupTime) > chillTime * 1000
+				|| (lastPopupMessage.MessageType != item.MessageType)
+				|| item.MessageType == CAN_POPUP_MSG_HANDBRAKE
+				|| item.MessageType
+						== CAN_POPUP_MSG_ENGINE_OIL_PRESSURE_FAULT_STOP_THE_VEHICLE) {
+
+			PushPopupMsg(&item, canSemaphore, popupMessageQueue);
+			previousCanPopupTime = millis();
+		}
+
+	}
+
+	void Process(unsigned long currentTime) {
+
+		if (GetEngineRunning() && !automaticLightingShownOnEngineRunning
+				&& automaticLightingShownOnIgnition) {
+			PushPopupMsg(&itemlight, canSemaphore, popupMessageQueue);
+			automaticLightingShownOnEngineRunning = true;
+		}
+
+		if (canBeVisible && !IsPopupVisible()
+				&& !popupMessageQueue->isEmpty()) {
+
+			xSemaphoreTake(canSemaphore, portMAX_DELAY);
+			popupMessageQueue->pop(&currentPopupMessage);
+			xSemaphoreGive(canSemaphore);
+
+			ShowCanPopupMessage(currentPopupMessage.Category, currentPopupMessage.MessageType, currentPopupMessage.KmToDisplay, currentPopupMessage.DoorStatus1, currentPopupMessage.DoorStatus2, currentPopupMessage.Counter);
+			lastPopupMessage = currentPopupMessage;
+		}
+
+		if (((millis() - canDisplayPopupStartTime) > CAN_POPUP_MESSAGE_TIME)
+				&& popupVisible) {
+			HideCanPopupMessage(currentPopupMessage.MessageType,
+					currentPopupMessage.DoorStatus1,
+					currentPopupMessage.Counter);
+
+			canDisplayPopupStartTime = 0;
+		}
+
+	}
+
+	void ShowCanPopupMessage(uint8_t category, uint8_t messageType,
+			int kmToDisplay, uint8_t doorStatus1, uint8_t doorStatus2,
+			int counter) {
+
+		CanDisplayPacketSender displayMessageSender(canMessageSender);
+		canDisplayPopupStartTime = millis();
+
+		popupVisible = true;
+		uint8_t messageSentCount = 0;
+
+		while (messageSentCount < CAN_POPUP_MESSAGE_SEND_COUNT) {
+			displayMessageSender.ShowPopup(category, messageType, kmToDisplay,
+					doorStatus1, doorStatus2);
+			messageSentCount++;
+
+			vTaskDelay(5 / portTICK_PERIOD_MS);
+		}
+
+	}
+
+	void ShowCanDoorPopUp() {
+		ShowCanPopupMessage(currentPopupMessage.Category,currentPopupMessage.MessageType, currentPopupMessage.KmToDisplay, currentPopupMessage.DoorStatus1,	currentPopupMessage.DoorStatus2, currentPopupMessage.Counter);
+	}
+
+	void HideCanPopupMessage(uint8_t messageType, uint8_t doorStatus, int counter) {
+		CanDisplayPacketSender displayMessageSender(canMessageSender);
```

Comment:
This could be a class level member which is instantiated at the constructor. This way the object is created every time we call this method. Actually this is also wrong in my implementation :/